### DISCUSSION TITLE
Regedit & wsl distribution management

### DISCUSSION THREAD

**Discussion by hbadi** [microsoft/WSL#7700] [Q&A]
Hi,

I manage wsl distributions in very particular way, i mean i do not export/import but i directly manage distros using the regedit (i just create manually a guid distro with all keys/values). The reason for that is imy distros are stored in an external disks and i need to use them from differents computers (desk/home), export/import take a very long time cause and vhdx file are self containing (why do i need to export). Unfortunatly sometimes wsl doesn't regonize hand-crafted distrubutions for no particular reason, only distrubutions created using the conventional way are remaining. I don't how to restore a distro from a vhdx file, any idea ?

Thanks for helping.

**Comment by Biswa96** [comment]
* First, trace the events from Microsoft.Windows.HyperV.Compute provider. You can find the GUID of that in this repo.
* Second, open a working distro in WSL2 and watch dmesg output. If anything goes wrong at init binary you can know it from there.
* Third, if nothing works you can mount the vhdx file with a Hyper-V Compute System API in a running WSL2 system as a SCSI device. If required I can share a sample code.

**Reply by hbadi** [reply]
Hi @Biswa96,

I think i broke my wsl, distros are are not listed anymore...cause i lauched `diskmgmt.msc` (a message asked me to do something related to gpt for unknown disks...). How can i now if a vhdx file is not corrupted or not used by anything else in windows ?

I tried to use lxrunoffline, now i can list my distros but also some old ones, and the result is not the same as `wsl -l`, strange... I believed wsl/lxrunoffline were only based on registry information, is there some others places where informations are stored (files for example) ?

Sorry it's a little bit confusing, let me know if you need more information.

Thanks.

**Reply by Biswa96** [reply]
> cause i lauched diskmgmt.msc (a message asked me to do something related to gpt for unknown disks...)

Than happen when the vhdx file is mounted in Windows somehow (generally by double clicking it). Don't mount it or format it, otherwise the files will be lost forever. I am not sure what you have done. And I am not familiar with the lxrunoffline project. But that project is a good one.

**Reply by hbadi** [reply]
Yes i double-clicked on vhdx files but nothing was mounted, i think the vhdx file is ok.

It's worst and worst, now i can't even reinstall a new distro, i get the following message :
`Wslregisterdistribution Failed With Error 0x80070050`

**Reply by Biswa96** [reply]
🤷‍♂️

**Reply by hbadi** [reply]
I did a whole clean with lxrunoffline, now i can install a brand new distro in which i look inside vhdx file using `file` command.
 It appears my original vhdx file is corrupt since it doesn't contains id signature : "NO Log Signature".
How can i fix that ?

**Reply by Biswa96** [reply]
I don't know how to fix that. Generally, I would convert the VHDX file to VHD using `Convert-VHD` cmdlet in PowerShell. Then extract the required files with 7zip from that VHD file.

**Reply by hbadi** [reply]
Hi @Biswa96 ,

I don't have `Convert-VHD` (i'm using a regular windows).
I found a way at least to restore some files using `dmde` software.
Conclusion for me is : not manupulating registry...

**Reply by Biswa96** [reply]
Though I am not suggesting to manipulate Lxss registry, it does not hurt the VHDX file.

**Reply by hbadi** [reply]
I agree, problems occurs when i tried to manipulate vhdx file (double-click in the explorer or use the disk mgnt utility). That's a pity i couldn't fix, i'm sure it's alsmost nothing to fix the partition table to get the the right enter point to boot (i made a comparison between a safe and a corrupted vhdx file, it's only about id signature).
I'd be great to have a real WSL swiss-knife gui superviser.
Anyway i'd like to thank you for your support, let me know if you get further tricks about registry/vhdx.
